components/clarity-repl/src/repl/datastore.rs

```diff
@@ -319,193 +325,316 @@ impl ClarityBackingStore for Datastore {
-            block_id_lookup,
-            open_chain_tip: id,
-            current_chain_tip: id,
-            chain_height: 0,
-            height_at_chain_tip,
             current_epoch: StacksEpochId::Epoch2_05,
             current_epoch_start_height: 0,
             constants,
-            genesis_time,
         }
     }
 
     pub fn get_current_epoch(&self) -> StacksEpochId {
         self.current_epoch
     }
 
-    pub fn get_current_block_height(&self) -> u32 {
-        self.chain_height
+    pub fn get_current_stacks_block_height(&self) -> u32 {
+        self.stacks_chain_height
     }
-    pub fn advance_chain_tip(&mut self, count: u32) -> u32 {
-        let cur_height = self.chain_height;
-        let current_lookup_id = *self
+
+    pub fn get_current_burn_block_height(&self) -> u32 {
+        self.burn_chain_height
+    }
+
+    fn build_next_stacks_block(&self, clarity_datastore: &ClarityDatastore) -> StacksBlockInfo {
+        let burn_chain_height = self.burn_chain_height;
+        let stacks_block_height = self.stacks_chain_height;
+
+        let last_stacks_block = self
+            .stacks_blocks
+            .get(&clarity_datastore.current_chain_tip)
+            .expect("current chain tip missing in stacks block table");
+        let last_burn_block = self
+            .burn_blocks
+            .get(&height_to_burn_block_header_hash(burn_chain_height))
+            .expect("burn block missing in burn block table");
+
+        let last_block_time = std::cmp::max(
+            last_stacks_block.stacks_block_time,
+            last_burn_block.burn_block_time,
+        );
+
+        let bytes = height_to_hashed_bytes(stacks_block_height);
+
+        let block_header_hash = {
+            let mut buffer = bytes;
+            buffer[0] = 1;
+            BlockHeaderHash(buffer)
+        };
+        let burn_block_header_hash = height_to_burn_block_header_hash(burn_chain_height);
+        let consensus_hash = {
+            let mut buffer = bytes;
+            buffer[0] = 3;
+            ConsensusHash::from_bytes(&buffer[0..20]).unwrap()
+        };
+        let vrf_seed = {
+            let mut buffer = bytes;
+            buffer[0] = 4;
+            VRFSeed(buffer)
+        };
+        let stacks_block_time: u64 = last_block_time + 10;
+
+        StacksBlockInfo {
+            block_header_hash,
+            burn_block_header_hash,
+            consensus_hash,
+            vrf_seed,
+            stacks_block_time,
+        }
+    }
+
+    pub fn advance_burn_chain_tip(
+        &mut self,
+        clarity_datastore: &mut ClarityDatastore,
+        count: u32,
+    ) -> u32 {
+        for _ in 1..=count {
+            let last_stacks_block = self
+                .stacks_blocks
+                .get(&clarity_datastore.current_chain_tip)
+                .unwrap();
+            let last_burn_block = self
+                .burn_blocks
+                .get(&last_stacks_block.burn_block_header_hash)
+                .unwrap();
```

Comment:
The use of `unwrap()` on `self.stacks_blocks.get()` and `self.burn_blocks.get()` means that the function will panic if these lookups fail.